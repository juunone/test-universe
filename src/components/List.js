import React from "react";
import PropTypes from "prop-types";

export const List = ({ data, deleteList }) => {
  return (
    <ul data-testid="todoList">
      {data.map((value, index) => {
        return (
          <li key={index} style={{ margin: "1em 0" }}>
            {value}
            <span
              style={{ cursor: "pointer" }}
              onClick={() => {
                deleteList(index);
              }}
            >
              X
            </span>
          </li>
        );
      })}
    </ul>
  );
};

List.propTypes = {
  data: PropTypes.array.isRequired,
  deleteList: PropTypes.func.isRequired
};
