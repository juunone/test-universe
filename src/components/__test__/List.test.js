import React from "react";
import { create, act } from "react-test-renderer";

import { List } from "../List";

describe("List", () => {
  let root;
  act(() => {
    root = create(<List data={["첫번째"]} deleteList={jest.fn()} />);
  });

  it("renders correctly", () => {
    expect(root.toJSON()).toMatchSnapshot();
  });

  it("should render as long as the data", () => {
    act(() => {
      root.update(<List data={["첫번째", "두번쨰"]} deleteList={jest.fn()} />);
    });

    expect(root.toJSON()).toMatchSnapshot();
  });
});
