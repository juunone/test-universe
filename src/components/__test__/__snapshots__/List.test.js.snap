// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders correctly 1`] = `
<ul
  data-testid="todoList"
>
  <li
    style={
      Object {
        "margin": "1em 0",
      }
    }
  >
    첫번째
    <span
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      X
    </span>
  </li>
</ul>
`;

exports[`List should render as long as the data 1`] = `
<ul
  data-testid="todoList"
>
  <li
    style={
      Object {
        "margin": "1em 0",
      }
    }
  >
    첫번째
    <span
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      X
    </span>
  </li>
  <li
    style={
      Object {
        "margin": "1em 0",
      }
    }
  >
    두번쨰
    <span
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      X
    </span>
  </li>
</ul>
`;
